### PR TITLE
ModelMapper for mapping fields of Category & Discount

### PR DESCRIPTION
**Pull request by VolodymyrBushko**

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/category/mapper/CategoryMapper.java [hunk 1]
```diff
@@ -3,28 +3,23 @@
 import com.exadel.discountwebapp.category.entity.Category;
 import com.exadel.discountwebapp.category.vo.CategoryRequestVO;
 import com.exadel.discountwebapp.category.vo.CategoryResponseVO;
+import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Component;
 
 @Component
 public class CategoryMapper {
 
+    private final ModelMapper modelMapper = new ModelMapper();
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/category/mapper/CategoryMapper.java:13]: add Strict strategy config to the mapper on the constructor level

File: src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java [hunk 2]
```diff
@@ -16,65 +17,29 @@
...
-                .price(request.getPrice())
-                .startDate(request.getStartDate())
-                .expirationDate(request.getExpirationDate())
-                .percentage(request.getPercentage())
-                .quantity(request.getQuantity())
-                .perUser(request.getPerUser())
-                .category(category)
-                .vendor(vendor)
-                .build();
+        modelMapper.getConfiguration().setAmbiguityIgnored(true);
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java]: probably can be removed?

File: src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java [hunk 3]
```diff
@@ -16,65 +17,29 @@
...
-        discount.setImageUrl(request.getImageUrl());
-        discount.setFlatAmount(request.getFlatAmount());
-        discount.setPrice(request.getPrice());
-        discount.setStartDate(request.getStartDate());
-        discount.setExpirationDate(request.getExpirationDate());
-        discount.setPercentage(request.getPercentage());
-        discount.setQuantity(request.getQuantity());
-        discount.setPerUser(request.getPerUser());
-        discount.setCategory(category);
         discount.setVendor(vendor);
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java:51]: add to skip policy?

File: src/main/java/com/exadel/discountwebapp/category/mapper/CategoryMapper.java (hunk 1 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/category/mapper/CategoryMapper.java:13]: done

File: src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java (hunk 2 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java]: removed

File: src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java (hunk 3 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java:51]: done

**Review by ahlukhouski2 — APPROVED** [review]

**Pull request merged**